File: wiki_java/src/main/java/com/github/klee0kai/wiki/provide/inject/FeatureScreen.java
package com.github.klee0kai.wiki.provide.inject;

import com.github.klee0kai.stone.Stone;
import com.github.klee0kai.wiki.provide.identifiers.FeaturePresenter;

import javax.inject.Inject;

public class FeatureScreen {

    @Inject
    public FeaturePresenter presenter;

    @Inject
    void init(FeaturePresenter presenter) {

    }

    void start() {
        AppComponent DI = Stone.createComponent(AppComponent.class);
        DI.inject(this);
    }

}

File: wiki_kotlin/src/main/kotlin/com/github/klee0kai/wiki/provide/inject/FeatureScreen.kt
package com.github.klee0kai.wiki.provide.inject

import com.github.klee0kai.wiki.provide.identifiers.FeaturePresenter
import javax.inject.Inject

class FeatureScreen {

    @Inject
    lateinit var presenter: FeaturePresenter

    @Inject
    fun init(presenter: FeaturePresenter) {
    }

    fun start() {
        val DI = AppComponentStoneComponent()
        DI.inject(this)
    }
}
